### PR TITLE
Add docs to kustomize chart

### PR DESCRIPTION
**Pull request by michaellzc**
This PR added example to integrate kustomize into `helm install`

Why not `helm-x`? I just realize the project was abandoned and doesn't work with helm v3. Its core functionality was extracted into another library https://github.com/variantdev/chartify. 

### Test plan

See added README.md

close https://github.com/sourcegraph/sourcegraph/issues/32178

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #62** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/62" target="_blank">Graphite</a> 
    * **PR #63** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/63" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: charts/sourcegraph/examples/kustomize-chart/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+This example demonstrate the escape hatch whenever you encounter something that our helm charts do not support yet. [Kustomize] allows you to apply [strategic merge patch] and [json patch] on helm chart without maintaining a custom fork. This is our **recommended** approach to customize helm chart, and you should avoid forking unless absolutely necessary.
+
+## Customize `frontend` service port
+
+In this example, we will change the `http` port of `sourcegraph-frontend.Service.yaml` and update the corresponding backend service port number in `sourcegraph-frontend.Ingress.yaml`.
+
+> Below command should be run within the current directory
+
+```sh
+helm upgrade --install --create-namespace -n sourcegraph sourcegraph sourcegraph/sourcegraph --post-renderer ./kustomize
```
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/examples/kustomize-chart/README.md:12]: Might be worth adding some extra explanation about the `./kustomize` script and its role in this process

File: charts/sourcegraph/examples/kustomize-chart/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# kustomize chart
+
+This example demonstrate the escape hatch whenever you encounter something that our helm charts do not support yet. [Kustomize] allows you to apply [strategic merge patch] and [json patch] on helm chart without maintaining a custom fork. This is our **recommended** approach to customize helm chart, and you should avoid forking unless absolutely necessary.
```
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/examples/kustomize-chart/README.md:3]: Maybe add an extra "if you do this, file a feature request" sentence?

**Review by caugustus-sourcegraph — APPROVED** [review]

File: charts/sourcegraph/examples/kustomize-chart/README.md (hunk 2 above)
Comment (michaellzc) [reply charts/sourcegraph/examples/kustomize-chart/README.md:3]: We should definitely do it. However, we haven't figured out how customers can provide helm feedback yet.

I'll start a thread on slack later today or tomorrow

**Review by caugustus-sourcegraph — APPROVED** [review]

**Comment by michaellzc** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).


Job Reference: 7AR5iRB1puWVsiuAXgLi

**Pull request merged**